projects/igniteui-angular/src/lib/grids/grid-base.component.ts

```diff
@@ -3486,6 +3486,12 @@ export abstract class IgxGridBaseComponent extends DisplayDensityBase implements
      * @memberof IgxGridBaseComponent
      */
     public addRow(data: any): void {
+        // If row editing is enabled and there is a row in edit mode
+        if (this.rowEditable && this.crudService.row) {
```

Comment:
Looking at this now I believe we should exit edit mode in all cases, regardless of if there is or there is no row edit enabled. So you can remove this check.